src/main/java/com/teragrep/cfe_39/consumers/kafka/HDFSWrite.java

```diff
@@ -0,0 +1,250 @@
+
+            // hack for running locally with fake DNS records
+            // set this to true if overriding the host name in /etc/hosts
+            conf.set("dfs.client.use.datanode.hostname", config.getKerberosTestMode());
+
+            // server principal
+            // the kerberos principle that the namenode is using
+            conf.set("dfs.namenode.kerberos.principal.pattern", config.getKerberosPrincipal());
+
+            // set usergroup stuff
+            UserGroupInformation.setConfiguration(conf);
+            UserGroupInformation.loginUserFromKeytab(config.getKerberosKeytabUser(), config.getKerberosKeytabPath());
+
+            // filesystem for HDFS access is set here
+            fs = FileSystem.get(conf);
+        }
+    }
+
+    // Method for committing the AVRO-file to HDFS
+    public void commit(File syslogFile, long lastEpochMicros) {
+        // The code for writing the file to HDFS should be same for both test (non-kerberized access) and prod (kerberized access).
+        if (useMockKafkaConsumer) {
+            // CODE FOR TEST-MODE GOES HERE!
+            //Get the filesystem - HDFS
+            try {
+                //==== Create directory if not exists
+                Path workingDir = fs.getWorkingDirectory();
+                // Sets the directory where the data should be stored, if the directory doesn't exist then it's created.
+                Path newDirectoryPath = new Path(path);
+                if (!fs.exists(newDirectoryPath)) {
+                    // Create new Directory
+                    fs.mkdirs(newDirectoryPath);
+                    LOGGER.info("Path <{}> created.", path);
+                }
+
+                //==== Write file
+                LOGGER.debug("Begin Write file into hdfs");
+                //Create a path
+                Path hdfswritepath = new Path(newDirectoryPath.toString() + "/" + fileName); // filename should be set according to the requirements: 0.12345 where 0 is Kafka partition and 12345 is Kafka offset.
+                if (fs.exists(hdfswritepath)) {
+                    throw new RuntimeException("File " + fileName + " already exists");
+                } else {
+                    LOGGER.info("Path <{}> doesn't exist.", path);
+                }
+
+                /*//Init output stream
+                FSDataOutputStream outputStream = fs.create(hdfswritepath);
+                // Write the file contents of syslogFile to hdfswritepath in HDFS.
+                // file to bytes[]
+
+                *//*byte[] bytearray = new byte[(int) syslogFile.length()];
+                try (FileInputStream inputStream = new FileInputStream(syslogFile)) {
+                    inputStream.read(bytearray);
+                }*//*
+                byte[] bytes = Files.readAllBytes(Paths.get(syslogFile.getPath())); // if readAllBytes is not efficient use FileInputStream
+                outputStream.write(bytes);
+                outputStream.close();*/
```

Comment:
Don't comment out code

Reply:
Solved in 022832c8b423efc7466b7aead56cc317a45a0002